### PR TITLE
Release/0.10.0

### PR DESCRIPTION
**Pull request by vanbasten17**
**Current changelog**

- Mobile responsive
- Added new customizable scrollbars (enabled by default in Windows)
- Export staticAsset in @botonic/react's index to make it available from bot's project
- Enabled attachments
- Messages bubbles (bot/user) pseudo-elements are now modified according to botMessageStyle or userMessageStyle 
- Customizable text input area
- Customizable typography (fontFamily) attribute
- Fixed typing indicator not showing up when message had delay/typing
- Now typing delay working in production (added new export in bot's `index.js`: `export const config = { defaultDelay: 1, defaultTyping: 2 }`)
- Removed console errors generated by @rebass/grid dep.
- Customizable styles for Button and Reply components
- New handoff options
- Carousels with one element don't break
- PersistentMenu, adding `{closeLabel: 'Your_copy'}` inside persistentMenu, will change the label

### CODE REVIEW

**Comment by vanbasten17** [comment]
# Changelog
All notable changes to Botonic will be documented in this file.

## [Unreleased]

## [0.10.0] - 2019-11-08
### Added
- Added changelog.
- Native support for mobile devices for Webchat. Customizable breakpoint (`mobileBreakpoint`, **webchat/index.js**).
- Enable media attachments for Webchat (`userInput.attachments.enable` of **webchat/index.js**).
- Enable an emoji picker for Webchat (`userInput.emojiPicker`, **webchat/index.js**).
- Customizable scrollbar for Webchat (`theme.scrollbar` , **webchat/index.js**). Now the default scrollbar's style will be like the one in OS X systems for all platforms (Mac, Linux and Windows).
- New handoff options for handling cases with [Hubtype's Desk](https://app.hubtype.com/). Refer to [Human Handoff](https://docs.botonic.io/main-concepts/human-handoff) section.
- Customization of Webchat's header, userInput bottom area, userInput text box, replies and buttons styles (`header.style`, `reply.style`, `button.style`, `userInput.style`, `userInput.box.style`, **webchat/index.js**).
- Cancel's label of the Webchat's persistent menu can now be changed by passing to `persistentMenu` the object with `{closeLabel: 'newCloseLabel'}`.
- `export const config = { defaultDelay: 1, defaultTyping: 2 }` can be added to **src/index.js** to define a global configuration for typing and delay options.

### Changed
- Start using "Changelog".
- Rewrite [Human Handoff](https://docs.botonic.io/main-concepts/human-handoff) section.
- Rewrite [Botonic Settings](https://docs.botonic.io/main-concepts/botonic-settings) section.
- Rewrite [Custom Webchat](https://docs.botonic.io/tutorials-and-examples/custom-webchat) section.
- Updated [custom-webchat template](https://github.com/hubtype/botonic/tree/master/packages/botonic-cli/templates/custom-webchat).

### Fixed
- Typography now will be changed in a proper way for all Webchat components by passing the `fontFamily` attribute in `theme.style` in **webchat/index.js**.
- Webchat's triangles of messages bubbles (bot/user) are now modified according to `style.background` passed to `botMessageStyle` or `userMessageStyle`.
- Typing indicator not showing up when messages had delay/typing as webchat didn't scroll to bottom.
- Now typing and delay work properly in production.
- Remove errors prompted in browser developer's console by changing dep `@rebass/grid` to `rebass`.
- Carousels with just one element won't be broken.
- Using `staticAsset` in `custom-webchat` template that produced a bug in production.

[Unreleased]: https://github.com/hubtype/botonic/compare/master...HEAD
[0.10.0]: https://github.com/hubtype/botonic/compare/master...release/0.10.0

File: packages/botonic-react/src/webchat/webchat.jsx [hunk 1]
```diff
@@ -74,16 +74,19 @@ export const Webchat = forwardRef((props, ref) => {
...
-        let themeProperty = null
-        if (getProperty(theme, v) == null || getProperty(theme, k) == null) {
-          themeProperty = null
-        } else {
-          themeProperty = getProperty(theme, v) || getProperty(theme, k)
-        }
-        return themeProperty
+        let nestedProperty = getProperty(theme, v)
+        let plainProperty = getProperty(theme, k)
+        if (nestedProperty !== undefined) return nestedProperty
```
Comment (dpinol) [inline packages/botonic-react/src/webchat/webchat.jsx]: pedantic: swap lines 86 & 87?

File: packages/botonic-react/src/webchat/webchat.jsx [hunk 2]
```diff
@@ -74,16 +74,19 @@ export const Webchat = forwardRef((props, ref) => {
   const [emojiPickerOpened, setEmojiPickerOpened] = useState(false)
   const [attachment, setAttachment] = useState({})
 
-  const getThemeProperty = property => {
+  /*
+   * returns the value of a property defined in bot's theme based on CUSTOM_WEBCHAT_PROPERTIES dictionary.
+   * it gives preference to nested defined properties over plain properties.
```
Comment (dpinol) [inline packages/botonic-react/src/webchat/webchat.jsx]: I think an example would make clear what nested and plain properties are

File: packages/botonic-react/tests/webchat.theme.test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        border: 'none',
+        color: 'blue',
+        borderRadius: '20px',
+      },
+    },
+  },
+  botMessageImage: null,
+}
+
+const getThemeProperty = (property, defaultValue = undefined) => {
```
Comment (dpinol) [inline packages/botonic-react/tests/webchat.theme.test.js]: you should never test a copy of the real code. 
You can move getThemeProperty to utils.js with getProperty, or move both to a new properties.js
to inject the them, one way is having a higher order function:
```
function getThemeProperty(theme) {
   return (property, defaultValue = undefined) => {
       ...
   }
}
```

File: packages/botonic-react/src/webchat/webchat.jsx (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-react/src/webchat/webchat.jsx]: changed 👍

File: packages/botonic-react/tests/webchat.theme.test.js (hunk 3 above)
Comment (vanbasten17) [reply packages/botonic-react/tests/webchat.theme.test.js]: now fixed and refactored, passing the test as expected 👍

**Pull request merged**